File: flutter/core/lib/src/storage/storage_service.dart
/// Key-value storage service contract.
abstract class StorageService {
  Future<void> write(String key, String value);
  Future<String?> read(String key);
  Future<void> delete(String key);
}

/// Stub in-memory implementation for scaffolding.
class StubStorageService implements StorageService {
  final _store = <String, String>{};

  @override
  Future<void> write(String key, String value) async => _store[key] = value;

  @override
  Future<String?> read(String key) async => _store[key];

  @override
  Future<void> delete(String key) async => _store.remove(key);
}

File: ios/Packages/Core/Sources/Core/Storage/StorageService.swift
import Foundation

/// Key-value storage service protocol.
public protocol StorageService: AnyObject {
    func write(key: String, value: String)
    func read(key: String) -> String?
    func delete(key: String)
}

/// Stub in-memory implementation for scaffolding.
public final class StubStorageService: StorageService {
    private var store: [String: String] = [:]
    public init() {}

    public func write(key: String, value: String) { store[key] = value }
    public func read(key: String) -> String? { store[key] }
    public func delete(key: String) { store.removeValue(forKey: key) }
}
